backend/server/routers/courses.py

```diff
@@ -91,8 +92,20 @@ def fixUserData(userData: dict):
     },
 )
 def getCourse(courseCode: str):
-    """get info about a course given courseCode"""
+    """get info about a course given courseCode
+    - start with the current database
+    - if not found, check the archives
+    """
     result = coursesCOL.find_one({"code": courseCode})
+
+    if not result:
+        years = sorted(ARCHIVED_YEARS, reverse=True)
```

Comment:
its probably not worth making this variable, you should probably make `sorted` part of the for loop directly. If youre worried about performance, `sorted` is a generator anyways, so the performance here is identical.